packages/eslint-plugin-gestalt/src/eslintASTHelpers.js

```diff
@@ -0,0 +1,233 @@
+  variableNode,
+}) => {
+  const keyValues = retrieveKeyValuesFromVariable({
+    context,
+    variableNode,
+  });
+
+  return buildPropsFromKeyValues({ keyValues });
+};
+
+/* ↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓  getComponenFromAttribute ↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓
+This function returns the component containing the attribute's node (nodeAttribute).
+Examples:
+<div {...props} /> >> returns div node for the spread props attribute
+*/
+type GetComponentFromAttributeType = ({| nodeAttribute: GenericType |}) => GenericType;
+
+export const getComponenFromAttribute: GetComponentFromAttributeType = ({ nodeAttribute }) =>
+  nodeAttribute.parent;
+
+/* ↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓  getVariableNodeInScopeFromName ↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓
+This function returns the component's name containing the attribute's node (nodeAttribute).
+Examples:
+<div {...props} /> >> returns div for the spread props attribute
+*/
+type GetVariableNodeInScopeFromNameType = ({|
+  context: GenericType,
+  nodeElement: GenericType,
+  name: string,
+|}) => GenericType;
+
+export const getVariableNodeInScopeFromName: GetVariableNodeInScopeFromNameType = ({
+  context,
+  nodeElement,
+  name,
+}) => {
+  const scope = context.getScope(nodeElement);
+  // Look in local scope for variable reference
+  const variableNode = scope.references.find((reference) => reference.identifier.name === name);
+  return variableNode;
+};
+
+/* ↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓  getComponentNameFromAttribute ↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓
+This function returns the component's name containing the attribute's node (nodeAttribute).
+Examples:
+<div {...props} /> >> returns div for the spread props attribute
+*/
+type GetComponentNameFromAttributeType = ({| nodeAttribute: GenericType |}) => string;
+
+export const getComponentNameFromAttribute: GetComponentNameFromAttributeType = ({
+  nodeAttribute,
+}) => nodeAttribute?.parent?.name?.name;
+
+/* ↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓  hasImport ↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓
+This function checks is a given node (importNode) contains a given import path (path), and returns true if so.
+Examples:
+import { Box } from 'gestalt' >> path="gestalt"
+import { Box } from 'app/box' >> path="app/box"
+*/
+type HasImportType = ({| importNode: GenericType, path: string |}) => boolean;
+
+export const hasImport: HasImportType = ({ importNode, path }) => {
+  const importName = importNode.source ? importNode.source.value : null;
+  return importName === path;
+};
+
+/* ↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓  updateGestaltImportFixer ↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓
+This function returns an array of arrays containing the named imports ([imported name, local or aliased name]) from a node (importNode).
+*/
+type GetNamedImportsComponentsType = ({| importNode: GenericType |}) => $ReadOnlyArray<
+  $ReadOnlyArray<string>,
+>;
+
+export const getNamedImportsComponents: GetNamedImportsComponentsType = ({ importNode }) => {
+  const namedImports = importNode.specifiers.map((node) => [node.imported.name, node?.local?.name]);
+  return namedImports;
+};
+
+/* ↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓  getHtmlTag ↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓
```

Comment:
@AlbertCarreras Could we use regular JSCode format instead?

```js
type IsTagType = ({| elementNode: GenericType, tagName: string |}) => boolean;

/**
 * This function checks is a given node (elementNode) contains a given tag (tagName), and returns true if so.
 * Examples:
 *   `<div />` if tagName="div" returns true
 *   `<div />` if tagName="button" returns false
 */
export const isTag: IsTagType = ({ elementNode, tagName }) => elementNode?.name?.name === tagName;
```

It actually works better with VSCode's hover capabilities:

![image](https://user-images.githubusercontent.com/127199/128602432-85421853-4dc7-4962-906d-ccf0edbd0d34.png)
![image](https://user-images.githubusercontent.com/127199/128602473-63639822-937c-4958-8189-283feaa219c8.png)

Reply:
Yeah... good point. DONE